### PR TITLE
use-cases: Refactoring

### PR DESCRIPTION
**Pull request by j-mueller**
Update all the contracts in `plutus-use-cases`:

* Pull the validation code out of the Template Haskell quotes
* Use record labels
* Add `traceErrorH` to the PlutusTx prelude and use it where possible
* Delete `Stage0` modules
* Infix use of `and`, `or`, `txSignedBy`

### CODE REVIEW

**Comment by j-mueller** [comment]
Looking at the diff I just noticed a few places that could be improved, I will ping @michaelpj when that's done.

File: plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/MultiSigStateMachine.hs [hunk 1]
```diff
@@ -46,19 +50,183 @@ import qualified Language.PlutusTx.StateMachine as SM
...
+
+mkValidator :: Params -> (State, Maybe Input) -> (State, Maybe Input) -> PendingTx -> ()
+mkValidator p ds vs ptx = 
+    let sm = StateMachine (stepWithChecks p ptx) stateEq in
+    SM.mkValidator sm ds vs ptx
+
+validator :: Params -> ValidatorScript
+validator params = 
+    ValidatorScript $ 
+        Ledger.applyScript
```
Comment (michaelpj) [inline plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/MultiSigStateMachine.hs]: I mildly think we should use `applyScript` infix throughout so the "function" comes first.

File: plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Swap.hs [hunk 2]
```diff
@@ -9,20 +8,29 @@ module Language.PlutusTx.Coordination.Contracts.Swap(
...
+PlutusTx.makeLift ''Ratio
+
+timesR :: Ratio Integer -> Ratio Integer -> Ratio Integer
+timesR (x :% y) (x' :% y') = (P.multiply x x') :% (P.multiply y y')
+
+plusR :: Ratio Integer -> Ratio Integer -> Ratio Integer
+plusR (x :% y) (x' :% y') = (P.plus (P.multiply x y') (P.multiply x' y)) :% (P.multiply y y')
+
+minusR :: Ratio Integer -> Ratio Integer -> Ratio Integer
+minusR (x :% y) (x' :% y') = (P.minus (P.multiply x y') (P.multiply x' y)) :% (P.multiply y y')
```
Comment (michaelpj) [inline plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Swap.hs]: Infix functions in these definitions?

**Comment by j-mueller** [comment]
> We should remember to update the playground use cases too.

Yeah they are still on the list, but I'm going to update the tutorials now since those are needed for the plutus book

File: plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/MultiSigStateMachine.hs (hunk 1 above)
Comment (j-mueller) [reply plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/MultiSigStateMachine.hs]: Yeah, and it saves a few parens

**Pull request merged**